### PR TITLE
[Feat] Automatically download dashboard through nuget

### PR DESCRIPTION
**Pull request by SaifAqqad**
- The download functionality will only be used if the workload is not installed.

- The runner will check if there are any matching dashboards previously downloaded and use them

- Otherwise, the runner will fetch the available dashboard versions from [nuget.org](https://www.nuget.org/packages?q=Aspire.Dashboard.Sdk) based on the running [RID](https://learn.microsoft.com/en-us/dotnet/core/rid-catalog)

- The runner will select the appropriate version to download based on the passed runtime version option or the latest installed runtime, passed option should be validated against installed runtimes and against available dashboard versions (major runtime version should match major dashboard version)

- The runner will extract the downloaded package in the data folder (.AspireRunner)

### CODE REVIEW

File: src/AspireRunner.Core/AspireDashboardOptions.cs [hunk 1]
```diff
@@ -104,6 +104,16 @@ public sealed class RunnerOptions
     /// </summary>
     [JsonConverter(typeof(JsonStringEnumConverter))]
     public SingleInstanceHandling SingleInstanceHandling { get; set; } = SingleInstanceHandling.WarnAndExit;
+
+    /// <summary>
```
Comment (WeeeeeeeeeeS) [inline src/AspireRunner.Core/AspireDashboardOptions.cs:108]: Good documentation
image

File: src/AspireRunner.Core/Helpers/DotnetCli.Regex.cs [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
-namespace AspireRunner.Core;
+namespace AspireRunner.Core.Helpers;
 
 public partial class DotnetCli
 {
     [GeneratedRegex(@"([\d\.]+)\s+(?:\[(.+)\])?", RegexOptions.Compiled)]
     private static partial Regex SdkOutputRegex();
 
+    [GeneratedRegex(@"(.+?) ([\d\-_.\w]+?) \[(.+)\]", RegexOptions.Compiled)]
+    private static partial Regex RuntimeOutputRegex();
```
Comment (WeeeeeeeeeeS) [inline src/AspireRunner.Core/Helpers/DotnetCli.Regex.cs:11]: give examples about regex

File: src/AspireRunner.Core/Helpers/NugetHelper.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+using NuGet.Packaging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace AspireRunner.Core.Helpers;
+
+public class NugetHelper(ILogger<NugetHelper> logger)
+{
+    private const string RepoUrl = "https://api.nuget.org/v3/index.json";
```
Comment (WeeeeeeeeeeS) [inline src/AspireRunner.Core/Helpers/NugetHelper.cs:12]: REPO_URL > RepoUrl

File: src/AspireRunner.Tool/Program.cs [hunk 4]
```diff
@@ -1,4 +1,5 @@
 ﻿using AspireRunner.Core;
+using AspireRunner.Core.Helpers;
```
Comment (WeeeeeeeeeeS) [inline src/AspireRunner.Tool/Program.cs:2]: imported without a use????

File: src/AspireRunner.Core/AspireDashboardOptions.cs (hunk 1 above)
Comment (SaifAqqad) [reply src/AspireRunner.Core/AspireDashboardOptions.cs:108]: image

File: src/AspireRunner.Core/Helpers/DotnetCli.Regex.cs (hunk 2 above)
Comment (SaifAqqad) [reply src/AspireRunner.Core/Helpers/DotnetCli.Regex.cs:11]: just ask gemini to explain it bro

File: src/AspireRunner.Core/Helpers/NugetHelper.cs (hunk 3 above)
Comment (SaifAqqad) [reply src/AspireRunner.Core/Helpers/NugetHelper.cs:12]: no

File: src/AspireRunner.Tool/Program.cs (hunk 4 above)
Comment (SaifAqqad) [reply src/AspireRunner.Tool/Program.cs:2]: its used on line 25 bro

just install rider i give u the account
tenor

File: src/AspireRunner.Core/Helpers/DotnetCli.Regex.cs (hunk 2 above)
Comment (WeeeeeeeeeeS) [reply src/AspireRunner.Core/Helpers/DotnetCli.Regex.cs:11]: oh yeah works too

File: src/AspireRunner.Tool/Program.cs (hunk 4 above)
Comment (WeeeeeeeeeeS) [reply src/AspireRunner.Tool/Program.cs:2]: it was a test if you're مصحصح or no, you passed

**Pull request merged**